src/DocChecks.jl

```diff
@@ -447,22 +447,36 @@ end
 
 function linkcheck(link::Base.Markdown.Link, doc::Documents.Document)
     if !haskey(doc.internal.locallinks, link)
-        const ERROR = typemax(Int)
-        local status =
-            try
-                local res = readstring(`curl -sI $(link.url)`)
-                local regex = r"^HTTP/1.1 (\d+) "
-                ismatch(regex, res) ? parse(Int, match(regex, res).captures[1]) : ERROR
-            catch err
-                ERROR
+        local result
+        try
+            result = readstring(`curl -sI $(link.url)`)
+        catch err
+            push!(doc.internal.errors, :linkcheck)
+            Utilities.warn("`curl -sI $(link.url)` failed:\n\n$(err)")
+            return false
+        end
+        local INDENT = " "^6
+        local STATUS_REGEX   = r"^HTTP/1.1 (\d+) (.+)$"m
+        if ismatch(STATUS_REGEX, result)
+            local status = parse(Int, match(STATUS_REGEX, result).captures[1])
+            if status < 300
+                print_with_color(:green, INDENT, "$(status) ", link.url, "\n")
+            elseif status < 400
+                local LOCATION_REGEX = r"^Location: (.+)$"m
+                if ismatch(LOCATION_REGEX, result)
+                    local location = strip(match(LOCATION_REGEX, result).captures[1])
+                    print_with_color(:yellow, INDENT, "$(status) ", link.url, "\n")
+                    print_with_color(:yellow, INDENT, " -> ", location, "\n\n")
```

Comment:
The indent here is to line up the two URLs so that it's easier to spot any differences. I played around with some different layouts and this one did seem to be the easiest to read to me.